administrator/models/user_profile.py

```diff
@@ -0,0 +1,92 @@
+"""
+    models.user_profile
+
+    This module contains model class for users.
+    """
+from django.contrib.auth.base_user import BaseUserManager
+from django.contrib.auth.models import AbstractBaseUser
+from django.contrib.auth.models import PermissionsMixin
+from django.contrib.auth.models import UserManager
+from django.core.mail import send_mail
+from django.db import models
+
+from django.utils.translation import ugettext_lazy as _
+
+class UserManager(BaseUserManager):
+    def _create_user(self, email, name, password, **extra_fields):
+        """Creates and saves a User with the given email, name and password."""
+
+        if not email:
+            raise ValueError('Users must have an email address')
+
+        email = self.normalize_email(email)
+        user = self.model(email=email, name=name, **extra_fields)
+        user.set_password(password)
+        user.save(using=self._db)
+        return user
+
+    def create_user(self, email, name, password, **extra_fields):
+        extra_fields.setdefault('is_superuser', False)
+        return self._create_user(email, name, password, **extra_fields)
+
+    def create_superuser(self, email, name, password, **extra_fields):
+        """Creates and saves a superuser with the given email, name and password."""
+
+        extra_fields.setdefault('is_superuser', True)
+        extra_fields.setdefault('is_staff', True)
+
+        if extra_fields.get('is_superuser') is not True:
+            raise ValueError('Superuser must have is_superuser=True.')
+
+        return self._create_user(email, name, password, **extra_fields)
+
+
+
+class UserProfile(AbstractBaseUser, PermissionsMixin):
+    """This class is used to handle data about users in DB.
+
+        A class implementing a fully featured User model with admin-compliant
+        permissions.
+
+        Email and password are required. Other fields are optional.
+    """
+
+    ACTIVE = 0
+    DELETED = 1
+    BANNED = 2
+    UNAUTHORIZED = 3
+
+    USER_STATUSES = (
+        (ACTIVE, 'active'),
+        (DELETED, 'deleted'),
+        (BANNED, 'banned'),
+        (UNAUTHORIZED, 'unauthorized'),
+    )
+    name = models.CharField(max_length=50, default='')
+    email = models.EmailField(_('email address'), unique=True, default='',
+                              error_messages={
+                                  'unique': _("A user with such email already exists."), })
```

Comment:
Please, refactor this according to PEP 8.
> Limit all lines to a maximum of 79 characters.
[https://www.python.org/dev/peps/pep-0008/#maximum-line-length](url)